### PR TITLE
Allow using Hash::get() with objects implementing ArrayAccess

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by chinpei215** [comment]
Ok, and I am sorry for my misreading. Indeed, passing an empty ArrayObject works correctly. 
Would it be a failling test case for is_array() check at line 74?
```php
$data = ['a' => [ 'b' => [ 'c' => ['d' => 1] ] ]];
$this->assertEquals(1, Hash::get(new ArrayObject($data), 'a.b.c.d'));
```

**Comment by ADmad** [comment]
@chinpei215 Yup it does fail. Thanks, i'll fix it.

**Comment by dereuromark** [comment]
:+1:

File: src/Utility/Hash.php [hunk 1]
```diff
@@ -43,7 +44,7 @@ class Hash
      * @return mixed The value fetched from the array, or null.
      * @link http://book.cakephp.org/3.0/en/core-libraries/hash.html#Hash::get
      */
-    public static function get(array $data, $path, $default = null)
+    public static function get($data, $path, $default = null)
```
Comment (markstory) [inline src/Utility/Hash.php]: What will happen when an object that doesn't implement ArrayAccess is passed?

Comment (ADmad) [inline src/Utility/Hash.php]: Nothing good I guess :)  Should I add a type check and throw exception if not an array or ArrayAccess instance?

Comment (markstory) [inline src/Utility/Hash.php]: I think so.

Comment (ADmad) [inline src/Utility/Hash.php]: Done

**Pull request merged**